### PR TITLE
feat(satp-hermes): improve audit endpoint

### LINKED ISSUE #4016: feat(satp-hermes): improve audit endpoint
**Is your feature request related to a problem? Please describe.**
The goal is to allow operators to 1) identify possible inconsistencies  from gateway crashes and 2) allowing to analyze transfers mediated by gateways.

Currently, the audit endpoint returns the stored sessions (in memory) that a gateway holds.
This ticket aims to:

1. Improve robustness of the audits by persisting audit data. Create a new type audit data "Audit" that the audit endpoint should return; Create a new SQL schema for and "AuditEntry". Persist in the local database audit information as sessions advance. An "AuditEntry" is an array of enhanced "SessionData". Enhanced session data is a session data object augmented with one or more bungee proofs. An "Audit" is a collection of "AuditEntry".
2. Refactor audit endpoint to retrieve and return audit data from the database
3. Add to each session object a set of corresponding BUNGEE views ("BUNGEEProof"), that serve as proofs of operations. While a type for a bungee proof is defined in the BUNGEE plugin, a schema needs to be created to persist it.
4. Improve available parameterization configurations to provide the audit endpoint (SPIKE). At the minimum it should be timestamps (already implemented); more include the ledgers the audit refers to;  and the human readable name of involved gateways (human readable names for identifying gateways are yet be implemented)
 
**Describe the solution you'd like**
-An updated endpoint with more parameters to scope audit
-Several schemas "Audit", "AuditEntry", "SessionData" (if necessary), "BUNGEEProof". 
-The AuditEntry should be created and persisted in run time; The audit endpoint should return Audit objects built from database data. 
-Tests showcasing both happy path and invalid transfer operations should be done; audits should be constructed from those tests such that it is possible to identify: 1) that the operation was well executed; 2) that the operation was not well executed, what was the problem, and what funds would be lost/in an inconsistent state.

### PR DESCRIPTION
**Pull request by pandaio22**
**Pull Request Requirements**
- [X] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [X] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [X] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [X] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [X] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

Closes #4016

### CODE REVIEW

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/interfaces/repository.ts [hunk 1]
```diff
@@ -166,6 +172,19 @@ export interface IRemoteLogRepository extends IRepository<RemoteLog, string> {
...
+  readById(auditId: string): Promise<AuditEntry>;
+  readByTimeInterval(
+    startTimestamp: number,
+    endTimestamp: number,
+  ): Promise<Audit>;
+  create(auditEntry: AuditEntry): any;
+  /** Clean up repository resources and connections */
+  destroy(): any;
+  /** Reset repository to initial state */
+  reset(): any;
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/interfaces/repository.ts:244]: please type the return values of the audit repository interface

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+   *
+   * Each AuditEntry represents a single transaction with its associated session data.
+   * This method inserts the entry into the audit_entries table (or logs table, depending on implementation)
+   * and preserves immutability; it does not modify existing entries.
+   *
+   * @param auditEntry - The transaction-level AuditEntry to persist, including transaction timestamp and sessions.
+   * @returns A promise resolving when the database insertion completes.
+   */
+  create(auditEntry: AuditEntry): any {
+    console.log(`Creating audit entry with ID: ${auditEntry.auditEntryId}`);
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts]: make sure audit logs are serialzied before persisting. please replace the console.log with the logger

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    const migrationSource = createMigrationSource();
+
+    config = {
+      ...config,
+      migrations: {
+        migrationSource: migrationSource,
+      },
+    } as Knex.Config;
+    console.log(
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts]: remove console.log; also do not print all the config (has passwords)

File: packages/cactus-plugin-satp-hermes/src/main/typescript/api1/admin/audit-endpoint.ts [hunk 4]
```diff
@@ -214,27 +242,42 @@ export class AuditEndpointV1 implements IWebServiceEndpoint {
...
-    } catch (ex) {
+    } catch (ex: any) {
       this.log.error(`Crash while serving ${reqTag}`, ex);
-      res.status(500).json({
-        message: "Internal Server Error",
-        error: ex?.stack || ex?.message,
+
+      const status = ex?.statusCode ?? 500;
+
+      res.status(status).json({
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/main/typescript/api1/admin/audit-endpoint.ts:278]: when error is 500, do not return traces, db errors, etc to the client; instead return a generic "internal error" message

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/knexfile-audit.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+ * ```
+ *
+ * @see {@link https://knexjs.org/guide/} Knex.js Documentation
+ * @see {@link https://www.postgresql.org/docs/} PostgreSQL Documentation
+ * @author Hyperledger Cacti Contributors
+ * @since 0.0.3-beta
+ */
+
+import path from "path";
+import { v4 as uuidv4 } from "uuid";
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/main/typescript/database/knexfile-audit.ts]: remove if not used

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+
+    const auditEntry: AuditEntry = {
+      auditEntryId: uuidv4(),
+      session: mockLocalLog,
+      timestamp: timestamp,
+    };
+    await repository.create(auditEntry);
+
+    const identity = gateway.Identity;
+    const startTimestamp = new Date(timestamp).toISOString();
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts]: better use a hardcoded time, as these two lines might introduce a race condition

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+    );
+
+    // Then
+    expect(auditResponse.status).toBe(200);
+    expect(auditResponse.data).toBeDefined();
+    expect(auditResponse.data.startTimestamp).toEqual(startTimestamp);
+    expect(auditResponse.data.endTimestamp).toEqual(endTimestamp);
+    expect(auditResponse.data.auditEntries).toBeDefined();
+    expect(auditResponse.data.auditEntries.entries.length).toBe(1);
+    expect(identity.gatewayServerPort).toBe(3010);
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts]: we do not need these assertions for the current functionality

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+    const invalidEndTimestamp = "ANOTHER_INVALID_TIMESTAMP";
+
+    try {
+      // When
+      await SATPGatewayAdminApi.performAudit(
+        invalidStartTimestamp,
+        invalidEndTimestamp,
+      );
+
+      fail("Request should have thrown AxiosError");
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts]: do not use fail; use expect instead

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+  afterEach(async () => {
+    await repository.getAuditEntriesTable().del();
+    if (gateway) {
+      await gateway.shutdown();
+    }
+  });
+
+  afterAll(async () => {
+    await repository.destroy();
+    fs.unlinkSync(dbPath);
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts]: this primitve call may fail

**Review by RafaelAPB — CHANGES_REQUESTED** [review]
Thank you for your contribution which will be invaluable for the SATP community. Left a few comments to be addressed

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/knexfile-audit.ts (hunk 5 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/database/knexfile-audit.ts]: Done

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts (hunk 3 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts]: Done

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts (hunk 2 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts]: Done

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts (hunk 9 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts]: Fixed

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts (hunk 7 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts]: Removed

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts (hunk 6 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts]: Fixed

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/interfaces/repository.ts (hunk 1 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/interfaces/repository.ts:244]: Fixed

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts (hunk 8 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-endpoint.test.ts]: Fixed

File: packages/cactus-plugin-satp-hermes/src/main/typescript/api1/admin/audit-endpoint.ts (hunk 4 above)
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/api1/admin/audit-endpoint.ts:278]: Fixed

**Comment by pandaio22** [comment]
@RafaelAPB Applied requested changes

File: packages/cactus-plugin-satp-hermes/src/main/yml/bol/oapi-api1.yml [hunk 10]
```diff
@@ -410,17 +410,17 @@ paths:
...
-            type: integer
-            format: int64
+            type: string
+            format: date-time
         - name: endTimestamp
           in: query
           description: The end timestamp for the audit period.
-          required: false
+          required: true
           schema:
```
[the remark this answers is not in the document]
Comment (RafaelAPB) [reply packages/cactus-plugin-satp-hermes/src/main/yml/bol/oapi-api1.yml:421]: We shouldn't need an end timestamp - if not provided it should default to "now"

**Comment by RafaelAPB** [comment]
@pandaio22 please review the comments by Copilot.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts [hunk 11]
```diff
@@ -618,6 +633,17 @@ export class SATPGateway implements IPluginWebService, ICactusPlugin {
       this.logger.info("Remote repository is not defined");
     }
 
+    if (this.config.auditRepository) {
+      this.auditRepository = new AuditEntryRepository(
+        this.config.auditRepository,
+      );
+    } else {
```
[the remark this answers is not in the document]
Comment (RafaelAPB) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts:653]: (this is a test) @copilot apply changes based on [this feedback](https://github.com/hyperledger-cacti/cacti/pull/4091#discussion_r3120402601)

File: packages/cactus-plugin-satp-hermes/src/main/yml/bol/oapi-api1.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (RafaelAPB) [reply packages/cactus-plugin-satp-hermes/src/main/yml/bol/oapi-api1.yml:421]: @pandaio22 testing copilot for quick fixes. @copilot apply changes based on [this feedback](https://github.com/hyperledger-cacti/cacti/pull/4091#discussion_r3120402612)

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/docker/satp-e2e-transfer-dev-dockerization-fast.test.ts [hunk 12]
```diff
@@ -909,6 +909,7 @@ describe("2 SATPGateways sending a token from Ethereum to Besu", () => {
     log.info("Amount was transfer correctly to the Owner account");
 
     // check audit endpoint and get audit data
+    /*
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/docker/satp-e2e-transfer-dev-dockerization-fast.test.ts]: Fixed. Removed dead code.

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/gateway/satp-e2e-transfer-1-gateway-with-api-server.test.ts [hunk 13]
```diff
@@ -57,22 +58,24 @@ const monitorService = MonitorService.createOrGetMonitorService({
...
-  }
-  if (ethereumEnv) {
-    await ethereumEnv.tearDown();
-  }
-  if (fabricEnv) {
-    await fabricEnv.tearDown();
-  }
+  await besuEnv.tearDown();
+  await ethereumEnv.tearDown();
+  await fabricEnv.tearDown();
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/gateway/satp-e2e-transfer-1-gateway-with-api-server.test.ts]: Fixed. Applied suggested changes.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+    this.logger.debug(
+      `Creating audit entry with ID: ${auditEntry.auditEntryId}`,
+    );
+
+    //return this.getAuditEntriesTable().insert(auditEntry);
+    return this.getAuditEntriesTable().insert({
+      auditEntryId: auditEntry.auditEntryId,
+      session: JSON.stringify(auditEntry.session),
+      timestamp: auditEntry.timestamp,
+    });
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts:184]: Fixed. Applied suggested code.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+   *
+   * @param auditEntryId - Unique audit entry identifier
+   * @returns Promise resolving to an audit entry or undefined
+   */
+  readById(auditEntryId: string): Promise<AuditEntry> {
+    return this.getAuditEntriesTable()
+      .where({ auditEntryId })
+      .first()
+      .then((row) => ({
+        auditEntryId: row.auditEntryId,
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/database/repository/knex-audit-repository.ts]: Fixed. It throws an `Audit entry with ID ${auditEntryId} not found` if no audit entry is found.

File: packages/cactus-plugin-satp-hermes/.gitignore [hunk 16]
```diff
@@ -9,6 +9,12 @@ src/main/solidity/generated/build-info/
 src/main/typescript/database/*
 !src/main/typescript/database/*.ts
 reports/**/*
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/.gitignore:11]: Fixed. Applied suggestion.

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/docker/satp-e2e-transfer-2-gateways-dockerization.test.ts [hunk 17]
```diff
@@ -711,6 +711,7 @@ describe("SATPGateway sending a token from Fabric to Besu", () => {
     log.info("Amount was transferred correctly to the Owner account");
 
     // check audit endpoint and get audit data
+    /*
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/docker/satp-e2e-transfer-2-gateways-dockerization.test.ts]: Fixed. Removed dead code.

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/gateway/satp-e2e-transfer-2-gateway-with-api-server.test.ts [hunk 18]
```diff
@@ -987,6 +987,7 @@ describe("2 SATPGateways sending a token from Besu to Ethereum", () => {
     log.info("Amount was transfer correctly to the Owner account");
 
     // check audit endpoint and get audit data
+    /*
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/gateway/satp-e2e-transfer-2-gateway-with-api-server.test.ts]: Fixed. Removed dead code.

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/gateway/satp-e2e-transfer-2-gateway-with-api-server.test.ts [hunk 19]
```diff
@@ -1313,7 +1315,8 @@ describe("2 SATPGateways sending a non fungible token from Besu to Ethereum", ()
       "Non fungible token was transferred correctly to the Owner account at Ethereum",
     );
 
-    // check audit endpoint and get audit data
+    // check audit endpoint and get audit
+    /*
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/gateway/satp-e2e-transfer-2-gateway-with-api-server.test.ts]: Fixed. Removed dead code.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/gateway-persistence.ts [hunk 20]
```diff
@@ -94,7 +102,13 @@ export class GatewayPersistence {
           sequenceNumber: logEntry.sequenceNumber,
         };
 
-        await this.storeInDatabase(localLog);
+        const auditEntry: AuditEntry = {
+          auditEntryId: `audit-${Date.now()}-${logEntry.sessionId}`,
+          session: localLog,
+          timestamp: Date.now(),
+        };
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/database/gateway-persistence.ts:110]: Fixed. Simplified to use uuid as the identifier.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/api1/admin/get-audit-handler-service.ts [hunk 21]
```diff
@@ -118,64 +122,33 @@ export async function executeAudit(
...
 
-    logger.info(
-      `${fnTag}, Adding session ${session.getSessionId()} to audit data.`,
-    );
-    sessionsData.push(session);
-  }
+  const audit: Audit = await auditRepository.readByTimeInterval(
+    Date.parse(new Date(req.startTimestamp).toISOString()),
+    Date.parse(new Date(req.endTimestamp).toISOString()),
+  );
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/api1/admin/get-audit-handler-service.ts]: Fixed. Added validation and error handling.

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-service.test.ts [hunk 22]
```diff
@@ -0,0 +1,143 @@
...
+    expect(entry.auditEntryId).toEqual(auditEntry.auditEntryId);
+    expect(entry.timestamp).toEqual(auditEntry.timestamp);
+    expect(entry.session).toEqual(auditEntry.session);
+  });
+
+  it("Given a invalid AuditRequest, When calling performAudit, Then it should return an empty AuditResponse", async () => {
+    // Given
+    // When
+    // Then
+  });
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-service.test.ts:195]: Fixed. Added more tests to cover edge cases.

File: packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-service.test.ts [hunk 23]
```diff
@@ -0,0 +1,143 @@
...
+  beforeEach(async () => {
+    await repository.getAuditEntriesTable().del();
+  });
+
+  afterAll(async () => {
+    if (gateway) {
+      await gateway.shutdown();
+    }
+    await repository.destroy();
+    fs.unlinkSync(dbPath);
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/test/typescript/integration/admin/satp-admin-audit-service.test.ts]: Fixed. Added suggested code.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/database/migrations/20260118174651_create_audit_entries_table.ts [hunk 24]
```diff
@@ -0,0 +1,26 @@
+import { Knex } from "knex";
+
+export function up(knex: Knex): Knex.SchemaBuilder {
+  return knex.schema.createTable("audit_entries", (table) => {
+    table.string("auditEntryId").notNullable().unique().primary();
+    table.string("session");
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/database/migrations/20260118174651_create_audit_entries_table.ts]: Fixed. Added suggested code.

**Review by pandaio22 — COMMENTED** [review]
> The audit repository connection is created here, but `SATPGateway.shutdown()` does not destroy it. This can leak DB connections and keep sqlite files locked. Consider registering a shutdown hook (or extending shutdown) to call `this.auditRepository.destroy()` when the gateway stops. 

Fixed. Added code to handle the auditRepository shutdown.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts:653]: Fixed. Added code to handle the auditRepository shutdown.

**Review by pandaio22 — COMMENTED** [review]
> We shouldn't need an end timestamp - if not provided it should default to "now" 

Fixed. Endtimestamp set as "optional". When not inserted, it defaults to now(). Add endpoint level test to validate this behavior.

File: packages/cactus-plugin-satp-hermes/src/main/yml/bol/oapi-api1.yml [hunk 25]
```diff
@@ -410,17 +410,17 @@ paths:
         - name: startTimestamp
           in: query
           description: The start timestamp for the audit period.
-          required: false
+          required: true
           schema:
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/yml/bol/oapi-api1.yml:414]: Fixed. Endtimestamp set as "optional". When not inserted, it defaults to now(). Added an endpoint level test to validate this behavior.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts [hunk 26]
```diff
@@ -683,8 +709,15 @@ export class SATPGateway implements IPluginWebService, ICactusPlugin {
         } else {
           throw new Error("GatewayIdentity is not defined");
         }
+        const hardcodedKnexConfig = {
+          client: "sqlite3",
+          connection: { filename: ":memory:" },
+          useNullAsDefault: true,
+        };
```
[the remark this answers is not in the document]
Comment (pandaio22) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts]: Note on OracleLogRepository initialization

If no oracleLogRepository is provided in config, we fall back to an in-memory SQLite instance (:memory:).

This approach was chosen to:

Avoid breaking existing tests and gateway instantiations
Ensure OraclePersistence always has a valid repository (no undefined handling)
Keep setup simple for dev/test environments

Trade-off:
In-memory storage is not persistent, so logs are lost on restart. This may be undesirable in production.

Future consideration:
We may restrict this fallback to test environments or require explicit configuration in production to avoid accidental data loss. I recommend for this to be discussed in an issue section and then opening a PR to implement the decided option.

File: packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts [hunk 27]
```diff
@@ -618,6 +646,31 @@ export class SATPGateway implements IPluginWebService, ICactusPlugin {
       this.logger.info("Remote repository is not defined");
     }
 
+    if (this.config.auditRepository) {
```
Comment (pandaio22) [inline packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts:649]: Note on OracleLogRepository initialization

If no oracleLogRepository is provided in config, we fall back to an in-memory SQLite instance (:memory:).

This approach was chosen to:

Avoid breaking existing tests and gateway instantiations
Ensure OraclePersistence always has a valid repository (no undefined handling)
Keep setup simple for dev/test environments

Trade-off:
In-memory storage is not persistent, so logs are lost on restart. This may be undesirable in production.

Future consideration:
We may restrict this fallback to test environments or require explicit configuration in production to avoid accidental data loss. I recommend for this to be discussed in an issue section and then opening a PR to implement the decided option.

**Comment by RafaelAPB** [comment]
@pandaio22 thank you for working on the feedback. The code looks good to me. The tradeoff makes sense - let's add a configuration  object in a next PR/documentation on SATP so operators know that they should initialize an oracleConfig. Please squash your commits into one and make sure that the codegen passes, prior to merge.

**Review by RafaelAPB — APPROVED** [review]

**Review by AndreAugusto11 — APPROVED** [review]
LGTM

**Review by LordKubaya — APPROVED** [review]
